universe/Species.cpp

```diff
@@ -356,85 +355,76 @@ SpeciesManager::SpeciesManager() {
     if (s_instance)
         throw std::runtime_error("Attempted to create more than one SpeciesManager.");
 
-    ScopedTimer timer("SpeciesManager Init", true, std::chrono::milliseconds(1));
-
-    try {
-        m_species = parse::species();
-    } catch (const std::exception& e) {
-        ErrorLogger() << "Failed parsing species: error: " << e.what();
-        throw e;
-    }
-
-    TraceLogger() << [this]() {
-            std::string retval("Species:");
-            for (const auto& pair : m_species) {
-                const auto& species = pair.second;
-                retval.append("\n\t" + species->Name() + "  \t");
-                retval.append(species->Playable() ?
-                                 "Playable " :
-                                 "         ");
-                retval.append(species->Native() ?
-                                 "Native " :
-                                 "       ");
-                retval.append(species->CanProduceShips() ?
-                                 "CanProduceShips " :
-                                 "                ");
-                retval.append(species->CanColonize() ?
-                                 "CanColonize " :
-                                 "            ");
-            }
-            return retval;
-        }();
-
     // Only update the global pointer on sucessful construction.
     s_instance = this;
 }
 
 const Species* SpeciesManager::GetSpecies(const std::string& name) const {
+    CheckPendingSpeciesTypes();
     auto it = m_species.find(name);
-    return it != m_species.end() ? it->second.get() : nullptr;
+    return it != end() ? it->second.get() : nullptr;
```

Comment:
why `end()` instead of `m_species.end()`? The extra call to `CheckPendingSpeciesTypes()` seems unnecessary.